### ISSUE TITLE
Bug: extra texture on fragment shader not working 

### ISSUE THREAD

**Issue by xavicolomer** [pixijs/pixijs#9632]
### Current Behavior

Shader

```
precision mediump float;

uniform sampler2D uSampler; // This is the default sampler that PIXI provides
uniform sampler2D sampleTexture; // This is your custom sampler
uniform float strength;

varying vec2 vTextureCoord; // This is the default varying texture coordinate from the vertex shader

void main(void) {
    // Sample from the original texture
    vec4 originalColor = texture2D(uSampler, vTextureCoord);

    // Sample from the custom sampler2D
    vec4 sampleColor = texture2D(sampleTexture, vTextureCoord);
    //sampleColor = vec4(1.0, 0.0, 0.0, 0.2);

    // Mix the colors based on the x-coordinate of the fragment.
    // Left half will be the original object and right half will be the custom sampler2D.
    float mixFactor = step(strength, vTextureCoord.x);
    vec4 mixedColor = mix(originalColor, sampleColor , mixFactor);

    gl_FragColor = mixedColor;
}
```

Code

```
const texture = await Assets.load(uniform.value)
filter.uniforms[uniform.name] = texture;
console.log(texture.valid);
```

texture.valid = true
if I uncomment sampleColor = vec4(1.0, 0.0, 0.0, 0.2); it works, a percentage of the image is red.

### Expected Behavior

I can see partially one image, and partially the other

### Steps to Reproduce

Use the code above

### Environment

- **`pixi.js` version**: *e.g. ^7.2.4*
- **Browser & Version**: *e.g. Chrome 108*
- **OS & Version**: *Mac*

**Comment by gfcrba** [comment]
Hi! You are trying to implement your Filter?
If so I bet you should override apply method to set your uniform in each frame.
From your code it seems to me that you do it only once.

**Comment by Zyie** [comment]
Hi there! 👋

We're doing some housekeeping and closing out older issues that haven't had recent activity.

PixiJS has evolved significantly since this issue was opened. If you're still experiencing this issue on PixiJS v8 (latest), please open a new issue with updated details and a minimal reproduction case.

Thanks for your understanding and for helping us keep the issue tracker focused and maintainable! 🩷
